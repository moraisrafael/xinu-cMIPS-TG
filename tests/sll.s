	.include "cMIPS.s"
	.text
	.align 2
	.set noat
	.globl _start
	.ent _start
_start: nop
	la $15,x_IO_BASE_ADDR
	la   $3,0x1
	nop
	nop
	nop
	sll $4,$3,1
	sw  $4, 4($15)  	# 2
	sll $4,$3,2
	sw  $4, 0($15)		# 4
	sll $4,$3,3
	sw  $4, 4($15)		# 8
	sll $4,$3,4
	sw  $4, 0($15)		# 16
	sll $4,$3,5
	sw  $4, 4($15)		# 32
	sll $4,$3,6
	sw  $4, 0($15)		# 64
	sll $4,$3,30
	sw  $4, 4($15)		# 1073741824 = 0x4000.0000
	sll $4,$3,31
	sw  $4, 0($15)		# -2147483648 = 0x8000.0000
	nop
	sw  $0, 0($15)
	sw  $0, 0($15)
	nop
	la  $15,x_IO_BASE_ADDR
	la  $8,0x01010101
	sll $4,$8,1
	sw  $4, 4($15)    # 33686018 = 02020202
	sll $4,$8,2
	sw  $4, 0($15)    # 67372036 = 04040404
	sll $4,$8,5
	sw  $4, 4($15)   # 538976288 = 20202020
	sll $4,$8,16
	sw  $4, 0($15)	  # 16842752 = 01010000
	sll $4,$8,31
	sw  $4, 4($15)   # -2147483648 = 80000000
	sw   $0, 0($15)
	sw   $0, 0($15)
	ori  $1,$0,31
	ori  $8,$0,1
	nop               # operandos rs e rt trocados!!
incr:	sllv $4,$8,$1     # R[rd] <- R[rt] << R[rs]  ==  $4 <- $8 << $1 
	sw   $4, 0($15) #-2147483648,134217728,8388608,524288,32768,2048,128,8
	addi $1,$1,-4
	slt  $22,$0,$1    # 0x8000.0000,x0800.0000,0x0080.0000,0x0008.0000,
	bne  $22,$0,incr  # 0x0000.8000,0x0000.0800,0x0000.0080,0x0000.0008
	nop
	wait
	.end _start
