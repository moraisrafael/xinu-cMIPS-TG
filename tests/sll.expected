00000002
00000004
00000008
00000010
00000020
00000040
40000000
80000000
00000000
00000000
02020202
04040404
20202020
01010000
80000000
00000000
00000000
80000000
08000000
00800000
00080000
00008000
00000800
00000080
00000008
